packages/core/contracts/financial-templates/common/financial-product-libraries/contract-for-difference-libraries/CoveredCallContractForDifferenceFinancialProductLibrary.sol

```diff
@@ -0,0 +1,72 @@
+ * - Say settlement price is 3500.  Then expiryPercentLong = (3500 - 3000) / 3500 = 0.143. The value of this 0.143 ETH
+ * is worth 0.143*3500=500 which is the percentage of the collateralPerPair that was above the strike price.
+ */
+contract CoveredCallContractForDifferenceFinancialProductLibrary is
+    ContractForDifferenceFinancialProductLibrary,
+    Lockable
+{
+    using FixedPoint for FixedPoint.Unsigned;
+    using SafeMath for uint256;
+
+    mapping(address => uint256) public contractForDifferenceStrikePrices;
+
+    /**
+     * @notice Enables any address to set the strike price for an associated CFD.
+     * @param contractForDifference address of the CFD.
+     * @param strikePrice the strike price for the covered call for the associated CFD.
+     * @dev Note: a) Any address can set the initial strike price b) A strike price cannot be 0.
+     * c) A strike price can only be set once to prevent the deployer from changing the strike after the fact.
+     * d) For safety, a strike price should be set before depositing any synthetic tokens in a liquidity pool.
+     * e) financialProduct must expose an expirationTimestamp method to validate it is correctly deployed.
+     */
+    function setContractForDifferenceParameters(address contractForDifference, uint256 strikePrice)
+        public
+        nonReentrant()
+    {
+        require(ExpiringContractInterface(contractForDifference).expirationTimestamp() != 0, "Invalid CFD address");
+        require(contractForDifferenceStrikePrices[contractForDifference] == 0, "Parameters already set");
+
+        contractForDifferenceStrikePrices[contractForDifference] = strikePrice;
+    }
+
+    /**
+     * @notice Returns a number between 0 and 1 to indicate how much collateral each long and short token are entitled
+     * to per collateralPerPair.
+     * @param expiryPrice price from the optimistic oracle for the CFD price identifier.
+     * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
+     */
+    function computeExpiryTokensForCollateral(int256 expiryPrice) public view override returns (uint256) {
+        uint256 contractStrikePrice = contractForDifferenceStrikePrices[msg.sender];
+
+        // If the expiry price is less than the strike price then the long options expire worthless (out of the money).
+        // Note we do not consider negative expiry prices in this call option implementation.
+        if (expiryPrice < 0 || uint256(expiryPrice) < contractStrikePrice)
+            return FixedPoint.fromUnscaledUint(0).rawValue;
+
+        // Else, token expires to be worth the fraction of a collateral token that's in the money. eg if ETH is $3500
+        // and strike is $3000, long token is redeemable for (3500-4000)/3500 = 0.143 WETH which is worth $500 and the
+        // short token is worth the remaining 0.8. This is strictly < 1, tending to 1 as the expiry tends to infinity.
```

Comment:
```suggestion
        // short token is worth the remaining 0.8. This is strictly < 1, tending to 1 as the expiry price tends to two times the strike price.
```

Isn't max payout 2* strike price?

Reply:
the max value for `expiryPercentLong` is 1 where all collateral in the CFD is allocated to the long side and 0 collateral is allocated to the short side. this will only happen if the price is infinity from `(expiryPrice - strikePrice) / expiryPrice`